src/services/address.service.ts

```diff
@@ -47,36 +101,143 @@ class AddressService {
+        return;
+      }
+      try {
+        let newRequest = { body: addressRequest.body.addresses[0] };
+
+        let address1 = await this.exact(newRequest);
+        let address2 = await this.exact({
+          body: addressRequest.body.addresses[1],
+        });
+
+        this.getDistance(
+          address1.latitude,
+          address1.longitude,
+          address2.latitude,
+          address2.longitude
+        )
+          .then((disances) => resolve(disances))
+          .catch((err) => {
+            loggerService
+              .error({
+                path: "/distance",
+                message: `${err.message + "in address.service"}`,
+              })
+              .flush();
+            reject(err);
+          });
+      } catch (err: any) {
+        loggerService
+          .warning({ message: err.message, path: "/address/distance" })
+          .flush();
+        reject(err);
+        return;
+      }
+    });
   }
 
-  // private async getDistance(lat1: string, lon1: string, lat2: string, lon2: string) {
-  //     // Defining this function inside of this private method means it's
-  //     // not accessible outside of it, which is perfect for encapsulation.
-  //     const toRadians = (degrees: string) => {
-  //         return degrees * (Math.PI / 180);
-  //     }
+  private async getDistance(
+    lat1: number,
+    lon1: number,
+    lat2: number,
+    lon2: number
+  ) {
+    // Defining this function inside of this private method means it's
+    // not accessible outside of it, which is perfect for encapsulation.
+    const toRadians = (degrees: number) => {
+      return degrees * (Math.PI / 180);
+    };
 
-  //     // Radius of the Earth in KM
-  //     const R = 6371;
+    // Radius of the Earth in KM
+    const R = 6371;
 
-  //     // Convert Lat and Longs to Radians
-  //     const dLat = toRadians(lat2 - lat1);
-  //     const dLon = toRadians(lon2 - lon1);
+    // Convert Lat and Longs to Radians
+    const dLat = toRadians(lat2 - lat1);
+    const dLon = toRadians(lon2 - lon1);
 
-  //     // Haversine Formula to calculate the distance between two locations
-  //     // on a sphere.
-  //     const a =
-  //         Math.sin(dLat / 2) * Math.sin(dLat / 2) +
-  //         Math.cos(toRadians(lat1)) * Math.cos(toRadians(lat2)) *
-  //         Math.sin(dLon / 2) * Math.sin(dLon / 2);
+    // Haversine Formula to calculate the distance between two locations
+    // on a sphere.
+    const a =
+      Math.sin(dLat / 2) * Math.sin(dLat / 2) +
+      Math.cos(toRadians(lat1)) *
+        Math.cos(toRadians(lat2)) *
+        Math.sin(dLon / 2) *
+        Math.sin(dLon / 2);
 
-  //     const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
 
-  //     // convert and return distance in KM
-  //     return R * c;
-  // }
+    // convert and return distance in KM & distance in MI
+    return [R * c, R * c * 0.62137119];
```

Comment:
Nice. I had assumed you would have added a second radians value with the circumference of the earth in miles. This looks cleaner, tbh.